daemon/pod.go

```diff
@@ -42,11 +43,201 @@ type Pod struct {
 	sync.RWMutex
 }
 
-func NewPod(rawSpec []byte, id string, data interface{}) (*Pod, error) {
+// TODO: remove convertToRunvPodSpec after pod.UserPod is deleted from runv
+func convertToRunvPodSpec(podSpec *apitypes.UserPod) (*pod.UserPod, error) {
```

Comment:
This PR adds `UserPod` in hyperd, so the dedicated `pod.UserPod` could be called `RunvPodSpec` and `pod.UserPod` should be removed in another PR. 

There is a todo to remove this convertion: `// TODO: remove convertToRunvPodSpec after pod.UserPod is deleted from runv`